python/lsst/ts/ess/controller/command_handler.py

```diff
@@ -0,0 +1,198 @@
+                callback_func=self._callback,
+                log=self.log,
+                disconnected_channel=None,
+            )
+            return device
+        elif device_configuration[common.Key.DEVICE_TYPE] == common.DeviceType.FTDI:
+            from .device import VcpFtdi
+
+            self.log.debug(
+                f"Creating VcpFtdi device with name {device_configuration[common.Key.NAME]} "
+                f"and sensor {sensor}"
+            )
+            device = VcpFtdi(
+                name=device_configuration[common.Key.NAME],
+                device_id=device_configuration[common.Key.FTDI_ID],
+                sensor=sensor,
+                callback_func=self._callback,
+                log=self.log,
+            )
+            return device
+        elif device_configuration[common.Key.DEVICE_TYPE] == common.DeviceType.SERIAL:
+            # make sure we are on a Raspberry Pi4
+            if "aarch64" in platform.platform():
+                from .device import RpiSerialHat
+
+                self.log.debug(
+                    f"Creating RpiSerialHat device with name {device_configuration[common.Key.NAME]} "
+                    f"and sensor {sensor}"
+                )
+                device = RpiSerialHat(
+                    name=device_configuration[common.Key.NAME],
+                    device_id=device_configuration[common.Key.SERIAL_PORT],
+                    sensor=sensor,
+                    callback_func=self._callback,
+                    log=self.log,
+                )
+                return device
+        raise RuntimeError(
+            f"Could not get a {device_configuration[common.Key.DEVICE_TYPE]!r} device"
+            f"on architecture {platform.platform()}. Please check the "
+            f"configuration."
+        )
+
+    def _get_sensor(self, device_configuration: dict) -> BaseSensor:
+        if device_configuration[common.Key.SENSOR_TYPE] == common.SensorType.HX85A:
+            sensor: BaseSensor = Hx85aSensor(
+                log=self.log,
+            )
+            return sensor
+        elif device_configuration[common.Key.SENSOR_TYPE] == common.SensorType.HX85BA:
+            sensor = Hx85baSensor(
+                log=self.log,
+            )
+            return sensor
+        elif (
+            device_configuration[common.Key.SENSOR_TYPE]
+            == common.SensorType.TEMPERATURE
+        ):
+            sensor = TemperatureSensor(
+                log=self.log,
+                num_channels=device_configuration[common.Key.CHANNELS],
+            )
+            return sensor
+        elif device_configuration[common.Key.SENSOR_TYPE] == common.SensorType.WIND:
+            sensor = WindSensor(
+                log=self.log,
+            )
+            return sensor
+        raise RuntimeError(
+            f"Could not get a {device_configuration[common.Key.SENSOR_TYPE]!r} sensor"
+            f"on architecture {platform.platform()}. Please check the "
```

Comment:
Does this method care about the architecture? If so, I missed it.

If it does not then please consider simplifying this message to remove anything about the architecture. (Unless it provides useful supplementary information)

Reply:
It doesn't so I simplified the message.